R/mig_resid.R

```diff
@@ -579,18 +584,20 @@ mig_resid_stock <- function(pop_m_mat,
     is.numeric(ages),
     is.numeric(ages_fertility)
   )
-  # Check in dimensions are ok
-  if(ncol(asfr_mat) == ncol(pop_f_mat)-1){
+  # Check in dimensions are ok - still working on this
+  if(ncol(asfr_mat) == ncol(pop_f_mat) -1 & nrow(sr_f_mat) == nrow(pop_f_mat) -1){
     print("matrix dimensions are correct")
+    else {
+    print("check matrix dimension")
   }
-  else if(ncol(sr_f_mat) == ncol(pop_m_mat) -1){
-    print("matrix dimensions are correct")
-  }
-  #if there are extra years, drop it
-  if(colnames(asfr_mat) == colnames(sr_f_mat)){
-    
-    
+  
+  #if there are extra years, drop it - still thinking the best way to deal with it
+  if(colnames(asfr_mat) != colnames(sr_f_mat)){
+    asfr_mat <- asfr_mat[, colnames(sr_f_mat)]
```

Comment:
This looks risky. You may consider instead splitting this into two parts. 1) get the year vectors & make sure the pre-filtered data objects have names, then use logic on the year vectors to see what's possible, making a determination, 2) then subset the data objects as needed.